### PR TITLE
✨  Upgrade to Go 1.22 and Kubernetes 1.29

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR updates the minimum required level of Go to 1.22, because that is the lowest that is supported now.

This PR updates the Kubernetes dependency to release 1.29, because we are no longer constrained by old OpenShift clusters.

This PR also updates the other dependencies as far as is allowed by the new Kubernetes version, except for open-cluster-management.io/addon-framework. _This_ PR only updates to release 0.9.3 of that. See #87 for an attempt to go to 0.10.0 --- which fails but I do not understand why.

This PR also updates the `make` target `deploy` to specify controller verbosity of 5, because `make deploy` is used for CI testing.

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/ocm/client.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func NewClient(config *rest.Config) (*client.Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	mapper, err := apiutil.NewDiscoveryRESTMapper(config, httpClient)
+	mapper, err := apiutil.NewDynamicRESTMapper(config, httpClient)
```
Comment (MikeSpreitzer) [inline pkg/ocm/client.go:24]: This change is required because the old function went away.

**Comment by waltforme** [comment]
Question: Where can we find the lowest supported version of Go?

**Comment by MikeSpreitzer** [comment]
I look at https://go.dev/dl/#stable

**Review by waltforme — APPROVED** [review]

**Pull request merged**